### PR TITLE
Fix spec class attribute for PersistenceStep and RefPixStep

### PR DESCRIPTION
**Pull request by jdavies-st**
~This PR fixes the comments in `JumpStep.spec` which were missing `#` before the comment.~

`PersistenceStep.spec` comments were also not formatted correctly and too long.

`RefPixStep.spec` comments were completely missing.  Pulled summaries of these from the docs.

Now all steps produce proper output when one does:

```
$ strun step_alias --help
```

Also a small change in `assign_wcs` picked up by the style checker

```
check-style: commands[0]> ruff .
jwst/assign_wcs/nirspec.py:404:8: E721 Do not compare types, use `isinstance()`
jwst/assign_wcs/nirspec.py:1828:8: E721 Do not compare types, use `isinstance()`
```

**Checklist for maintainers**
- [ ] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: jwst/jump/jump_step.py [hunk 1]
```diff
@@ -32,20 +32,20 @@ class JumpStep(Step):
...
-        extend_min_area = integer(default=90) Min area of emission after convolution for the detection of showers
-        extend_inner_radius = float(default=1) Inner radius of the ring_2D_kernel used for convolution
-        extend_outer_radius = float(default=2.6) Outer radius of the ring_2D_Kernel used for convolution
-        extend_ellipse_expand_ratio = float(default=1.1) Expand the radius of the ellipse fit to the extended emission
-        time_masked_after_shower = float(default=15) Seconds to flag as jump after a detected extended emission
-        max_extended_radius = integer(default=200) The maximum radius of an extended snowball or shower
-        minimum_groups = integer(default=3) The minimum number of groups to perform jump detection
-        minimum_sigclip_groups = integer(default=100) The minimum number of groups to switch to sigma clipping
-        only_use_ints = boolean(default=True) In sigclip only compare the same group across ints, if False compare all groups
+        extend_snr_threshold = float(default=1.2) # The SNR minimum for detection of extended showers in MIRI
```
Comment (hbushouse) [inline jwst/jump/jump_step.py:39]: #7778 also contains some pending updates to the jump_step spec block. So we'll need to make sure the two different sets of changes play nicely with one another.

File: jwst/refpix/refpix_step.py [hunk 2]
```diff
@@ -17,12 +17,12 @@ class RefPixStep(Step):
     class_alias = "refpix"
 
     spec = """
-        odd_even_columns = boolean(default=True)
-        use_side_ref_pixels = boolean(default=True)
-        side_smoothing_length = integer(default=11)
-        side_gain = float(default=1.0)
-        odd_even_rows = boolean(default=True)
-        ovr_corr_mitigation_ftr = float(default=1.8)
+        odd_even_columns = boolean(default=True) # Compute reference signal separately for even/odd columns (NIR only)
```
Comment (hbushouse) [inline jwst/refpix/refpix_step.py:20]: The config parser doesn't like having ()'s in the comment (it thinks it's another arg definition). Look at all the CI failures.

File: jwst/assign_wcs/nirspec.py [hunk 3]
```diff
@@ -401,7 +401,7 @@ def get_open_slits(input_model, reference_files=None, slit_y_range=[-.55, .55]):
     """
     exp_type = input_model.meta.exposure.type.lower()
     lamp_mode = input_model.meta.instrument.lamp_mode
-    if type(lamp_mode) == str:
+    if isinstance(lamp_mode, str):
```
[the remark this answers is not in the document]
Comment (jdavies-st) [reply jwst/assign_wcs/nirspec.py]: Done

File: jwst/jump/jump_step.py (hunk 1 above)
Comment (jdavies-st) [reply jwst/jump/jump_step.py]: Conflicts resolved

**Review by hbushouse — APPROVED** [review]
LGTM

**Pull request merged**